migrations/versions/201608241100_2ce1756a2f12_add_abstract_tables.py

```diff
@@ -0,0 +1,271 @@
+"""Add abstracts related tables
+
+Revision ID: 2ce1756a2f12
+Revises: 5596683819c9
+Create Date: 2016-08-22 14:57:28.005919
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from indico.core.db.sqlalchemy import UTCDateTime, PyIntEnum
+from indico.modules.events.abstracts.models.abstracts import AbstractState
+from indico.modules.events.abstracts.models.persons import AuthorType
+from indico.modules.events.abstracts.models.reviews import AbstractAction
+from indico.modules.users.models.users import UserTitle
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = '2ce1756a2f12'
+down_revision = '5596683819c9'
+
+
+def upgrade():
+    op.drop_constraint('fk_contributions_abstract_id_abstracts', 'contributions', schema='events')
+    op.drop_constraint('fk_abstract_field_values_abstract_id_abstracts', 'abstract_field_values',
+                       schema='event_abstracts')
+    op.drop_constraint('fk_judgments_abstract_id_abstracts', 'judgments', schema='event_abstracts')
+
+    op.rename_table('abstracts', 'legacy_abstracts', schema='event_abstracts')
+    op.execute('''
+        ALTER INDEX event_abstracts.ix_abstracts_accepted_track_id RENAME TO ix_legacy_abstracts_accepted_track_id;
+        ALTER INDEX event_abstracts.ix_abstracts_accepted_type_id RENAME TO ix_legacy_abstracts_accepted_type_id;
+        ALTER INDEX event_abstracts.ix_abstracts_event_id RENAME TO ix_legacy_abstracts_event_id;
+        ALTER INDEX event_abstracts.ix_abstracts_type_id RENAME TO ix_legacy_abstracts_type_id;
+        ALTER SEQUENCE event_abstracts.abstracts_id_seq RENAME TO legacy_abstracts_id_seq;
+        ALTER TABLE event_abstracts.legacy_abstracts RENAME CONSTRAINT pk_abstracts TO pk_legacy_abstracts;
+        ALTER TABLE event_abstracts.legacy_abstracts RENAME CONSTRAINT
+            fk_abstracts_accepted_type_id_contribution_types TO fk_legacy_abstracts_accepted_type_id_contribution_types;
+        ALTER TABLE event_abstracts.legacy_abstracts RENAME CONSTRAINT
+            fk_abstracts_event_id_events TO fk_legacy_abstracts_event_id_events;
+        ALTER TABLE event_abstracts.legacy_abstracts RENAME CONSTRAINT
+            fk_abstracts_type_id_contribution_types TO fk_legacy_abstracts_type_id_contribution_types;
+        ALTER TABLE event_abstracts.legacy_abstracts RENAME CONSTRAINT
+            uq_abstracts_friendly_id_event_id TO uq_legacy_abstracts_friendly_id_event_id;
+    ''')
+
+    op.create_table(
+        'abstracts',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('friendly_id', sa.Integer(), nullable=False),
+        sa.Column('event_id', sa.Integer(), nullable=False, index=True),
+        sa.Column('track_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('type_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('title', sa.String(), nullable=False),
+        sa.Column('description', sa.Text(), nullable=False),
```

Comment:
OTOH, we are gonna need at some point some way to consistently define an arbitrary number of rich text columns + properties.